### ISSUE TITLE
explanation?

### ISSUE THREAD

**Issue by smsaladi** [remram44/twitter_colormap_bot#4]
Could you explain how the second method in your jupyter notebook works please? I'm looking for some clarification on how the a minimum spanning tree and pruning procedure helps identify the colormap in an image.

**Comment by remram44** [comment]
Hi @smsaladi!

This piece of code was never actually used for the bot. The idea was to detect more accurately if an image is a heatmap or not.

To do that, we assume that a heatmap will exhibit a gradient of colors, varying continuously in a curve across the color space. We are not interested in the minimum spanning tree, but only that curve, so we prune leaves from the tree to get a single chain. If the image's colors are from a heatmap, it would be a chain of points along the curve in color space that is our colormap.

Note that this only really works for heatmaps, not for other plots that have less continuous colors along the colormap.

Once we have the chain, we know it's a colormap, we can compare it to a given colormap, and we should even be able to extract the colormap from an image.

---

I never actually deployed that to the bot, the hope was to stop classifying random images as plots just because they used two colors from a known colormap. The colorspace transformation is definitely a good idea (RGB is terrible), and I tried to do some learning from images I captured after running the bot for a while, but I never got to it.

At this point a lot fewer people are using Jet, so I stopped using the bot.
